src/dom-updates.js

```diff
@@ -0,0 +1,141 @@
+  lastFocusedElement = document.activeElement;
+  focusableElements[0].focus();
+  focusableElements.forEach(link => link.addEventListener('keydown', handleToggleEscape))
+};
+
+const handleToggleEscape = (e) => {
+  if (e.keyCode === 9 && document.activeElement.id === 'user-logout') {
+    lastFocusedElement.click();
+  }
+
+  if (e.keyCode === 9 && e.shiftKey && document.activeElement.id === 'user-settings') {
+    lastFocusedElement.click();
+  }
+
+  if (e.keyCode === 27) {
+    lastFocusedElement.click();
+  }
+};
+
+const updateNightsStayed = () => {
+  const nightsStayed = getUserBookings(todaysDate, currentUser.id, bookings, 'past');
+  totalNights.innerText = `${nightsStayed.length}`;
+};
+
+const updateTotalSpent = () => {
+  const nightsStayed = getUserBookings(todaysDate, currentUser.id, bookings, 'past');
+  const amoutSpent = getTotalSpent(nightsStayed, rooms);
+  totalSpent.innerText = `${amoutSpent}`;
+};
+
+const populateBookings = (date, bookings, rooms) => {
+  displayRooms.innerHTML = '';
+  // const formatedDate = formatDate(date);
+  // let timelineBookings;
+
+  // if (timeline === 'past') {
+  //   timelineBookings = bookings.filter(booking => formatedDate > booking.date);
+  // } else {
+  //   timelineBookings = bookings.filter(booking => formatedDate <= booking.date);
+  // }
+  
+  bookings.forEach(booking => {
+      const room = rooms.find(room => room.number === booking.roomNumber);
+
+      displayRooms.innerHTML += `
+      <article class="rooms">
+        <img class="room-image" src="./images/turing-logo.png" alt="turing logo">
+        <div class="room-info">
+          <h3 class="room-type">${room.roomType}</h3>
+          <p class="bed-size">${room.numBeds} ${room.bedSize}${room.bidet ? ', Bidet' : '' }</p>
+          <ul class="amenities">
+            <li>Wifi</li>
+            <li>Air conditioner</li>
+            <li>Balcony</li>
+            <li>Pet Friendly</li>
+            <li>Access to gym and pool</li>
+          </ul>
+          <p class="booked-date">Date: ${booking.date}</p>
+          <p class="reference">Booking Reference: ${booking.id}</p>
+          <p class="room-cost">$${room.costPerNight}</p>
+        </div>
+      </article>
+      `;
+  });
+};
+
+const populateAvailableRooms = (date, bookings, rooms) => {
+  displayRooms.innerHTML = '';
+  const formatedDate = formatDate(date)
+  const unavailableRooms = bookings.filter(booking => booking.date === formatedDate);
+
+  rooms.forEach(room => {
+    if(unavailableRooms.find(bookedRoom => bookedRoom.Number.includes(room.roomNumber))) {
+      displayRooms.innerHTML += `
+      <article class="rooms">
+        <img class="room-image" src="./images/turing-logo.png" alt="turing logo">
+        <div class="room-info">
+          <h3 class="room-type">${room.roomType}</h3>
+          <p class="bed-size">${room.numBeds} ${room.bedSize}${function () { if(room.bidet) { ', Bidet' } } }</p>
```

Comment:
Actually, this does not work. Although I found an article saying you can make a conditional work by wrapping it in an ES5 function it didn't work as intended. You have to use a tertiary operator to add the conditional in the string interpolation. Forgot to do it in this line since I was not using this function yet. Check out the populateBookings function.